thirdparty-api/transaction-patterns-linking.md

```diff
@@ -0,0 +1,403 @@
+<h1>Contents</h1>
+
+<!-- TOC -->
+
+- [1. Linking](#1-linking)
+  - [1.1. Pre-linking](#11-pre-linking)
+  - [1.2. Discovery](#12-discovery)
+  - [1.3. Request consent](#13-request-consent)
+      - [1.3.1. Web](#131-web)
+      - [1.3.2. OTP / SMS](#132-otp--sms)
+  - [1.4. Authentication](#14-authentication)
+      - [1.4.1. Web](#141-web)
+          - [DFSPAuthorizeSimulator](#dfspauthorizesimulator)
+      - [1.4.2. OTP](#142-otp)
+          - [DFSPAuthorizeSimulator](#dfspauthorizesimulator-1)
+  - [1.5. Grant consent](#15-grant-consent)
+  - [1.6. Credential registration](#16-credential-registration)
+      - [1.6.1. Deriving the challenge](#161-requesting-a-challenge)
+      - [1.6.2. Registering the credential](#162-registering-the-credential)
+      - [1.6.2. Finalizing the Consent](#163-finalizing-the-consent)
+- [2. Unlinking](#2-unlinking)
+- [3. Third-party credential registration](#3-third-party-credential-registration)
+  - [3.1. Authentication](#31-authentication)
+  - [3.2. Credential registration](#32-credential-registration)
+- [4. Linking Error Scenarios](#4-linking-error-scenarios)
+  - [4.1. Discovery](#41-Discovery)
+  - [4.2. Bad consentRequests](#42-Bad-consentRequests)
+  - [4.3. Authentication](#43-Authentication)
+  - [4.4. Grant consent](#44-Grant-consent)
+  
+<!-- /TOC -->
+
+# 1. Linking
+
+The goal of the linking process is to explain how users establish trust between
+all three interested parties:
+
+1. User
+2. DFSP where User has an account
+3. PISP that User wants to rely on to initiate payments
+
+*NOTE: For now, we're focusing on the FIDO / Web authentication channel.*
+
+Linking is broken down into several separate phases:
+
+1. **Pre-linking**
+   In this phase, a PISP asks what DFSPs are available to link with.
+2. **Request consent**
+   In this phase, a PISP attempts to establish trust between the 3 parties.
+3. **Authentication**
+   In this phase, a User proves their identity to their DFSP.
+4. **Grant consent**
+   In this phase, a PISP proves to the DFSP that the User and PISP have
+   established trust and, as a result, the DFSP confirms that mutual trust
+   exists between the 3 parties.
+5. **Credential registration**
+   In this phase, a User establishes the credential they'll use to consent to
+   future transfers from the DFSP and initiated by the PISP.
+
+## 1.1. Pre-linking
+
+In this phase, a PISP Server needs to know what DFSPs are available to link
+with. This is *unlikely* to be done on-demand (e.g., when a User clicks "link"
+in the PISP mobile App), and far more likely to be done periodically and cached
```

Comment:
Since this is an Async API, I guess we could also consider a case where the Switch sends a `PUT /services/THIRD_PARTY_DFSP` call to all PISPs whenever a new participant is added.